### PR TITLE
clientv3: use retryConnection

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: clientv3/client.go [hunk 1]
```diff
@@ -128,14 +128,15 @@ func (c *Client) retryConnection(oldConn *grpc.ClientConn, err error) *grpc.Clie
 	}
 	if oldConn != c.conn {
 		// conn has already been updated
-		return c.conn
+		return c.conn, nil
 	}
 	conn, dialErr := c.cfg.RetryDialer(c)
 	if dialErr != nil {
 		c.errors = append(c.errors, dialErr)
+		return nil, err
```
Comment (heyitsanthony) [inline clientv3/client.go]: should this be `dialErr`?

File: clientv3/kv.go [hunk 2]
```diff
@@ -87,30 +88,44 @@ type Txn interface {
...
+func (kv *kv) do(op Op) (*pb.ResponseUnion, error) {
+	for {
+		var err error
+		switch op.t {
+		// TODO: handle other ops
+		case tRange:
+			var resp *pb.RangeResponse
+			// TODO: setup sorting
+			r := &pb.RangeRequest{Key: op.key, RangeEnd: op.end, Limit: op.limit, Revision: op.rev}
+			for {
```
Comment (heyitsanthony) [inline clientv3/kv.go]: why loop here?

File: clientv3/kv.go [hunk 3]
```diff
@@ -87,30 +88,44 @@ type Txn interface {
 }
 
 type kv struct {
+	conn   *grpc.ClientConn // conn in-use
```
Comment (heyitsanthony) [inline clientv3/kv.go]: you can avoid storing the connection by using `client.activeConnection()`

Comment (xiang90) [inline clientv3/kv.go]: but activeConn might not be the one KV is using. so we might end up switch a good conn when there are multiple concurrent failures.

File: clientv3/kv.go (hunk 2 above)
Comment (xiang90) [inline clientv3/kv.go]: forgot to remove this... i moved for to line 106 :P

File: clientv3/kv.go (hunk 3 above)
Comment (heyitsanthony) [inline clientv3/kv.go]: ok

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**